presto-spi/src/main/java/com/facebook/presto/spi/plan/ProjectNode.java

```diff
@@ -74,14 +71,23 @@ public PlanNode getSource()
     }
 
     @Override
-    public <R, C> R accept(InternalPlanVisitor<R, C> visitor, C context)
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context)
     {
         return visitor.visitProject(this, context);
     }
 
     @Override
     public PlanNode replaceChildren(List<PlanNode> newChildren)
     {
-        return new ProjectNode(getId(), Iterables.getOnlyElement(newChildren), assignments);
+        requireNonNull(newChildren, "newChildren list is null");
+        checkArgument(newChildren.size() == 1, "newChildren list has multiple items");
+        return new ProjectNode(getId(), newChildren.get(0), assignments);
+    }
+
+    private static void checkArgument(boolean test, String errorMessage)
+    {
```

Comment:
This code is duplicated all over the place in SPI. Right now I copied convention and duplicated it again however, there should be a single static class or something that takes care of argument/state validation.

Here is a list of existing files with this code in SPI (they might be slightly different in behavior from one another but it's near identical)

1. com.facebook.presto.spi.plan.TopNNode
2. com.facebook.presto.spi.plan.LimitNode
3. com.facebook.presto.spi.plan.TableScanNode

And now:

1. ProjectNode.java
2. Assignments.java

Let me know if this is a worthwhile refactor or not.

Reply:
Check comment https://github.com/prestodb/presto/pull/13045#issuecomment-508021391

Reply:
Yes, please use duplicated private methods.

Reply:
Yeah I understand that. However, I feel like since then, the code has been duplicated in more places. Does it make sense to put it in PlanNode itself then? Since the functionality is duplicated in several subclasses.